docs/docs/ProgrammingGuide/keras-support.md

```diff
@@ -0,0 +1,108 @@
+# **Keras Support**
+
+BigDL supports loading pre-defined Keras models and running the models in a distributed manner.
+
+The Keras version we support and test is [__Keras 1.2.2__](https://faroit.github.io/keras-docs/1.2.2/).
+
+## **Load a Keras model into BigDL**
+
+A Keras model definition in __JSON__ file can be loaded as a BigDL model.
+Saved weights in __HDF5__ file can also be loaded together with the architecture of a model.
+See [here](https://faroit.github.io/keras-docs/1.2.2/getting-started/faq/#how-can-i-save-a-keras-model) on how to save the architecture and weights of a Keras model.
+
+You can directly use the API `load_keras` to load the Keras model into BigDL.
+
+__Remark__: Packages `tensorflow`, `keras==1.2.2` and `h5py` are required. They can be installed via `pip` easily.
+
+```python
+from bigdl.nn.layer import *
+
+bigdl_model = Model.load_keras(def_path, weights_path=None, by_name=False)
+```
+Parameters:
+
+* `def_path` The JSON file path containing the keras model definition to be loaded.
+* `weights_path`  The HDF5 file path containing the pre-trained keras model weights. Default to be `None` if you choose not to load weights. In this case, initialized weights will be used for the model.
+* `by_name`  Whether to load the weights of layers by name. Use this option only when you load the pre-trained weights. Default to be `False`, meaning that  weights are loaded based on the network's execution order topology. Otherwise, if it is set to be `True`, only those layers with the same name will be loaded with weights.
+
+## **Example**
+
+Here we illustrate with a concrete [LeNet example](https://github.com/fchollet/keras/blob/1.2.2/examples/mnist_cnn.py) from Keras.
+
+```python
+from keras.models import Sequential
+from keras.layers import Dense, Dropout, Activation, Flatten
+from keras.layers import Convolution2D, MaxPooling2D
+
+# Define a convnet model in Keras
+keras_model = Sequential()
+keras_model.add(Convolution2D(32, 3, 3, border_mode='valid',
+                              input_shape=(1, 28, 28)))
+keras_model.add(Activation('relu'))
+keras_model.add(Convolution2D(32, 3, 3))
+keras_model.add(Activation('relu'))
+keras_model.add(MaxPooling2D(pool_size=(2, 2)))
+keras_model.add(Dropout(0.25))
+
+keras_model.add(Flatten())
+keras_model.add(Dense(128))
+keras_model.add(Activation('relu'))
+keras_model.add(Dropout(0.5))
+keras_model.add(Dense(10))
+keras_model.add(Activation('softmax'))
```

Comment:
We should use `LogSoftMax` together with `ClassNLL`, or we can refactor this later after this PR (https://github.com/intel-analytics/BigDL/pull/1944 )merged